### DISCUSSION TITLE
fn:KeyNameIsColorant is a non-requirement

### DISCUSSION THREAD

**Discussion by bdoubrov** [pdf-association/arlington-pdf-model#60] [General] [answered]
There are no formal restrictions in PDF syntax on the values of colorant names. So, any Name object may potentially be a colorant name. As far as I can see, this predicate is used only for Type5 Halftone dictionary, where it gets a particular meaning as one of the key values of the **Halftone5Type** object. Would it make sense to replace this predicate by an explicit condition referencing this object?

**Comment by petervwyatt** [comment, marked as answer]
There are explicitly stated requirements in ISO 32000-2 in subclause 10.6.5.6 related to this but from your perspective - possibly, since you are purely syntax checking without a specific device model. 

However if you modelled a specific device with a non-standard explicit set of process colorants then there is additional information external to the PDF that you might be able to check.
